xtra-libp2p/src/endpoint_monitor.rs

```diff
@@ -0,0 +1,268 @@
+use crate::ConnectionStats;
+use crate::Endpoint;
+use crate::GetConnectionStats;
+use async_trait::async_trait;
+use libp2p_core::Multiaddr;
+use libp2p_core::PeerId;
+use std::collections::HashSet;
+use std::time::Duration;
+use tokio_tasks::Tasks;
+use xtra::message_channel::MessageChannel;
+use xtra::Address;
+use xtra_productivity::xtra_productivity;
+use xtras::SendInterval;
+
+/// How often we check for connection changes in the endpoint
+pub const CHECK_INTERVAL: Duration = Duration::from_secs(5);
+
+/// All subscribers to the connection monitor
+pub struct Subscribers {
+    connection_established: Vec<Box<dyn MessageChannel<ConnectionsEstablished>>>,
+    connection_dropped: Vec<Box<dyn MessageChannel<ConnectionsDropped>>>,
+    listen_address_added: Vec<Box<dyn MessageChannel<ListenAddressAdded>>>,
+    listen_address_removed: Vec<Box<dyn MessageChannel<ListenAddressRemoved>>>,
+}
+
+/// Convenience factory function to create Subscribers struct
+pub fn subscribers(
```

Comment:
🛠 I don't see the point of using a factory function here. Why not 

```
impl Subscribers {
 pub fn new(...) -> Self {
    ...
  }
}
```

Or even just making the fields of `Subscribers` `pub` since we always have to set all of them?

Reply:
I would make a constructor / factory function as this makes it easier to produce - otherwise, the likely mismatch in names will make this look really awful in endpoint_monitor constructor :( 

Happy to convert it into a `new()` function!